app/controllers/hiring_staff/vacancies/job_location_controller.rb

```diff
@@ -0,0 +1,76 @@
+class HiringStaff::Vacancies::JobLocationController < HiringStaff::Vacancies::ApplicationController
+  include FirstStepFormConcerns
+
+  before_action :redirect_unless_school_group_user_flag_on
+  before_action :set_up_url
+  before_action only: %i[create update] do
+    set_up_form(JobLocationForm)
+  end
+
+  def show
+    if @vacancy.present?
+      @form = JobLocationForm.new(@vacancy.attributes)
+    elsif session[:vacancy_attributes].present?
+      @form = JobLocationForm.new(session[:vacancy_attributes])
+    else
+      @form = JobLocationForm.new(school_group_id: current_school_group.id)
+    end
+  end
+
+  def create
+    store_vacancy_attributes(@form.vacancy.attributes)
+
+    if @form.complete_and_valid?
```

Comment:
This method is only necessary at the minute because of the way we handle dates. So both here and in `JobSpecificationController` we should be able to use @form.valid? It might be worth adding a comment to the `complete_and_valid?` method explaining that (or you can leave that to me)

Reply:
Added comment `# This method is only necessary for forms with specific error messages for date inputs.` Does that sound right?
(Couldn't sleep)